src/sql/SqlService.ts

```diff
@@ -0,0 +1,420 @@
+        }
+
+        if (sqlStatementOptions.hasOwnProperty('returnRawResult')){
+            tryGetBoolean(sqlStatementOptions.returnRawResult);
+        }
+    }
+
+    /**
+     * Converts datetime related wrapper classes to string to be able to serialize them. (No default serializers yet)
+     * @internal
+     * @param value
+     */
+    convertToStringIfDatetimeValue(value: any) {
+        if (value instanceof HzLocalTime || value instanceof HzLocalDate || value instanceof HzLocalDateTime) {
+            return value.toString();
+        } else if (value instanceof HzOffsetDateTime) {
+            return value.toISOString();
+        } else {
+            return value;
+        }
+    }
+
+    executeStatement(sqlStatement: SqlStatement): SqlResult {
+        try {
+            SqlServiceImpl.validateSqlStatement(sqlStatement);
+        } catch (error) {
+            throw new IllegalArgumentError(`Invalid argument given to execute(): ${error.message}`, error)
+        }
+
+        const connection = this.connectionRegistry.getRandomConnection(true);
+        if (connection === null) {
+            throw new HazelcastSqlException(
+                this.connectionManager.getClientUuid(),
+                SqlErrorCode.CONNECTION_PROBLEM,
+                'Client is not currently connected to the cluster.'
+            );
+        }
+
+        const queryId = SqlQueryId.fromMemberId(connection.getRemoteUuid());
+
+        const expectedResultType: SqlExpectedResultType =
+            sqlStatement.options?.expectedResultType ? SqlExpectedResultType[sqlStatement.options.expectedResultType]
+                : SqlServiceImpl.DEFAULT_EXPECTED_RESULT_TYPE;
+
+        let timeoutMillis: Long;
+        if(sqlStatement.options?.timeoutMillis){
+            if(typeof sqlStatement.options.timeoutMillis === 'number'){
+                timeoutMillis = Long.fromNumber(sqlStatement.options.timeoutMillis);
+            }else timeoutMillis = sqlStatement.options.timeoutMillis;
+        }else timeoutMillis = SqlServiceImpl.DEFAULT_TIMEOUT;
+
+        try {
+            const serializedParams = [];
+            if (Array.isArray(sqlStatement.params)) { // params can be undefined
+                for (let param of sqlStatement.params) {
+                    param = this.convertToStringIfDatetimeValue(param);
+                    serializedParams.push(this.serializationService.toData(param));
+                }
+            }
+            const cursorBufferSize = sqlStatement.options?.cursorBufferSize ?
+                sqlStatement.options.cursorBufferSize : SqlServiceImpl.DEFAULT_CURSOR_BUFFER_SIZE;
+            const requestMessage = SqlExecuteCodec.encodeRequest(
+                sqlStatement.sql,
+                serializedParams,
+                timeoutMillis,
+                cursorBufferSize,
+                sqlStatement.options?.schema || sqlStatement.options?.schema === '' || sqlStatement.options?.schema === null ?
```

Comment:
Since the `DEFAULT_SCHEMA` is `null`, can't we simplify this with `sqlStatement.options?.schema`  instead of this ternary if logic?

Reply:
`sqlStatement.options?.schema` is undefined if options is undefined, not null. So directly using `sqlStatement.options?.schema` does not work. We can remove `sqlStatement.options?.schema === null` check, but I wanted to be explicit. Going to remove it.

Reply:
Since the CodecUtil encodes `undefined` and `null` as a `NULL_FRAME`, I think just the thing I wrote will be enough.

Reply:
I think we should use default schema, I have a test that makes sure defaults are used if an option is not provided. So, I think

`sqlStatement.options?.hasOwnProperty('schema') ? sqlStatement.options.schema : SqlServiceImpl.DEFAULT_SCHEMA`

is ok, no need to over simplify.

you can search test with `should call encodeRequest with correct params`